h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/remote/RemoteXGBoostHandler.java

```diff
@@ -0,0 +1,117 @@
+package hex.tree.xgboost.remote;
+
+import hex.genmodel.utils.IOUtils;
+import hex.schemas.XGBoostExecReqV3;
+import hex.schemas.XGBoostExecRespV3;
+import hex.tree.xgboost.exec.LocalXGBoostExecutor;
+import hex.tree.xgboost.exec.XGBoostExecReq;
+import hex.tree.xgboost.task.XGBoostSaveMatrixTask;
+import org.apache.log4j.Logger;
+import water.Key;
+import water.api.Handler;
+import water.api.StreamingSchema;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class RemoteXGBoostHandler extends Handler {
+
+    private static final Logger LOG = Logger.getLogger(RemoteXGBoostHandler.class);
+
+    private static final Map<Key, LocalXGBoostExecutor> REGISTRY = new HashMap<>();
```

Comment:
there is no synchronization, does this need to be a concurrent hashmap? (NBHM)?

Reply:
good catch, fixed